FRENCH/src/ch10-02-traits.md

```diff
@@ -0,0 +1,1190 @@
+
+<!--
+```rust,ignore
+fn some_function<T: Display + Clone, U: Clone + Debug>(t: &T, u: &U) -> i32 {
+```
+-->
+
+```rust,ignore
+fn some_function<T: Display + Clone, U: Clone + Debug>(t: &T, u: &U) -> i32 {
+```
+
+<!--
+we can use a `where` clause, like this:
+-->
+
+... nous pouvons utiliser l'instruction `where`, comme ceci :
+
+<!--
+```rust,ignore
+fn some_function<T, U>(t: &T, u: &U) -> i32
+    where T: Display + Clone,
+          U: Clone + Debug
+{
+```
+-->
+
+```rust,ignore
+fn some_function<T, U>(t: &T, u: &U) -> i32
+    where T: Display + Clone,
+          U: Clone + Debug
+{
+```
+
+<!--
+This function’s signature is less cluttered: the function name, parameter list,
+and return type are close together, similar to a function without lots of trait
+bounds.
+-->
+
+La signature de cette fonction est moins encombrée : le nom de la fonction, la
+liste des paramètres, et le type de retour sont plus proches l'un de l'autre,
+comme une fonction sans traits liés.
+
+<!--
+### Returning Types that Implement Traits
+-->
+
+### Retourner des types qui implémentent des traits
+
+<!--
+We can also use the `impl Trait` syntax in the return position to return a
+value of some type that implements a trait, as shown here:
+-->
+
+Nous pouvons aussi utiliser la syntaxe `impl Trait` à la place du retour afin
+de retourner une valeur d'un type qui implémente un trait, comme ci-dessous :
+
+<!--
+```rust,ignore
+{{#rustdoc_include ../listings-sources/ch10-generic-types-traits-and-lifetimes/no-listing-05-returning-impl-trait/src/lib.rs:here}}
+```
+-->
+
+```rust,ignore
+{{#rustdoc_include ../listings/ch10-generic-types-traits-and-lifetimes/no-listing-05-returning-impl-trait/src/lib.rs:here}}
+```
+
+<!--
+By using `impl Summary` for the return type, we specify that the
+`returns_summarizable` function returns some type that implements the `Summary`
+trait without naming the concrete type. In this case, `returns_summarizable`
+returns a `Tweet`, but the code calling this function doesn’t know that.
+-->
+
+En utilisant `impl Resumable` pour le type de retour, nous indiquons que la
+fonction `retourne_resumable`retourne un type qui implémente le trait
+`Resumable` sans avoir à écrire le nom du type concret. Dans notre cas,
+`retourne_resumable` retourne un `Tweet`, mais le code qui appellera cette
+fonction ne le saura pas.
+
+<!--
+The ability to return a type that is only specified by the trait it implements
+is especially useful in the context of closures and iterators, which we cover
+in Chapter 13. Closures and iterators create types that only the compiler knows
+or types that are very long to specify. The `impl Trait` syntax lets you
+concisely specify that a function returns some type that implements the
+`Iterator` trait without needing to write out a very long type.
+-->
+
+La capacité de retourner un type qui est uniquement caractérisé par le trait
+qu'il implémente est tout particulièrement utile dans le cas des fermetures et
```

Comment:
The word `fermeture` seems wrong to me.

I won't provide any suggestion for now because I guess its use in lots of other places in the book. And it's in the terms translations section.